lib/select.c

```diff
@@ -385,9 +385,9 @@ int Curl_poll(struct pollfd ufds[], unsigned int nfds, timediff_t timeout_ms)
     if(ufds[i].fd == CURL_SOCKET_BAD)
       continue;
     if(ufds[i].revents & POLLHUP)
-      ufds[i].revents |= POLLIN;
+      ufds[i].revents |= (POLLRDNORM|POLLIN);
```

Comment:
Please elaborate. Why is setting these bits necessary?

Reply:
On Linux POLLPRI is equivalent to POLLRDNORM according to the [poll man page](https://www.man7.org/linux/man-pages/man2/poll.2.html), but the [kernel source code](https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/poll.h) actually shows different values for these. For example [on Windows](https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-wsapoll) "The POLLIN flag is defined as the combination of the POLLRDNORM and POLLRDBAND flag values.". So you are right, in this case it seems unnecessary to set both, I just thought it would be more explicit since there may be platforms defining it the other way around.

Reply:
Sorry, I meant POLLIN and not POLLPRI in my comment before this one. Also, I thought it would make semantically more sense to check for both values.

Reply:
I disagree. If this doesn't address a known problem, then I think this is just wrong. It actually adds a risk that something will change when we don't want it. We have after all managed without those bits for a very very long time on numerous operating systems.

Reply:
Okay, for this and the next line change I am fine with leaving it as is, but for the code below simulating poll via select, returning POLLRDNORM with POLLIN, POLLWRNORM with POLLOUT and POLLRDBAND with POLLPRI is just the correct thing to do as far as I understand the values poll should return for the corresponding socket states. Or am I wrong about that?

Reply:
Now moved into separate PR https://github.com/curl/curl/pull/5883.